### PR TITLE
chore(dotcom): refine forbidden vs not authorized error handling

### PR DESCRIPTION
**Pull request by mimecuvalo**
followup to the convo here: https://github.com/tldraw/tldraw/pull/4775
i'll leave this open so we can discuss our opinions!

### Change type

- [x] `other` 

### Test plan

1. Navigate to a forbidden or unauthorized route in the dotcom app and verify the error screen behavior.

- [ ] Unit tests
- [ ] End to end tests

### Release notes

- Refined error handling for forbidden and unauthorized states on the website.

### CODE REVIEW

File: apps/dotcom/client/src/components/StoreErrorScreen.tsx [hunk 1]
```diff
@@ -47,8 +47,8 @@ export function StoreErrorScreen({ error }: { error: Error }) {
 				return <LoginRedirectPage />
 			}
 			case TLSyncErrorCloseEventReason.FORBIDDEN: {
-				header = 'Not authorized'
-				message = 'You do not have permission to view this room.'
+				header = 'Invite only'
+				message = `Your account doesn't have permission to view this room.`
```
Comment (ds300) [inline apps/dotcom/client/src/components/StoreErrorScreen.tsx:51]:
```suggestion
				message = `You don't have permission to view this room.`
```

File: apps/dotcom/client/src/routes.tsx [hunk 2]
```diff
@@ -42,8 +42,8 @@ export const router = createRoutesFromElements(
 						)
 					}
 					case TLSyncErrorCloseEventReason.FORBIDDEN: {
-						header = 'Not authorized'
-						para1 = 'You do not have permission to view this file.'
+						header = 'Invite only'
+						para1 = `Your account doesn't have permission to view this room.`
```
Comment (ds300) [inline apps/dotcom/client/src/routes.tsx:46]:
```suggestion
						para1 = `You don't have permission to view this room.`
```

**Review by ds300 — APPROVED** [review]
I like 'Invite only'. But 'your account' is overly specific here.

**Comment by mimecuvalo** [comment]
ok cool 🤝

**Pull request merged**